ast/src/main/java/com/graphicsfuzz/common/typing/ScopeTrackingVisitor.java

```diff
@@ -90,7 +90,11 @@ public void visitForStmt(ForStmt forStmt) {
     //
     // because it declares 'i' twice in the same scope.
     //
-    // We thus have to push a new scope before traversing a 'for' statement and pop it afterwards.
+    // Furthermore, a 'while' statement introduces a new scope regardless of whether its body is
+    // a block or a single statement.
+    //
+    // We thus have to push a new scope before traversing a 'for' statement and pop it afterwards,
+    // and we do this regardless of whether the body is a block.
     pushScope();
     super.visitForStmt(forStmt);
     popScope();
```

Comment:
Did you mean to talk about "while" here?

Reply:
No - good catch, thanks!